group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -1812,23 +1844,36 @@ private CoordinatorResult<ConsumerGroupHeartbeatResponseData, CoordinatorRecord>
             .setClassicMemberMetadata(null)
             .build();
 
-        boolean bumpGroupEpoch = hasMemberSubscriptionChanged(
+        // If the group is empty, we must bump the epoch to fully
+        // intitialize the group.
```

Comment:
both cannot be empty. the issue was that when a member uses a regex and joins a new group, we must advance from epoch 0 (the default) and initialize the group. this only impacts the regex path because we don’t compute a new target assignment immediately.